pkg/query-service/app/querier/helper.go

```diff
@@ -208,14 +292,30 @@ func (q *querier) runBuilderQuery(
 		Name:   queryName,
 		Series: mergedSeries,
 	}
-	// Cache the seriesList for future queries
-	if len(missedSeries) > 0 && !params.NoCache && q.cache != nil {
-		mergedSeriesData, err := json.Marshal(mergedSeries)
+
+	var mergedSeriesData []byte
+	missedSeriesLen := len(missedSeries)
+	if missedSeriesLen > 0 && !params.NoCache && q.cache != nil {
+		// caching the data
+		var err error
+		mergedSeriesData, err = json.Marshal(mergedSeries)
 		if err != nil {
 			zap.S().Error("error marshalling merged series", zap.Error(err))
 			return
 		}
-		err = q.cache.Store(cacheKey, mergedSeriesData, time.Hour)
+	}
+
+	// response doesn't need everything
+	filterCachedPoints(mergedSeries, params.Start, params.End)
+	ch <- channelResult{
+		Err:    nil,
+		Name:   queryName,
+		Series: mergedSeries,
+	}
```

Comment:
There is additional `ch <- channelResult ` that is not removed above in this function, why?